Adding in a prompt so TypeSpec+Copilot users can add Go emitter configurations

richardpark-msft opened the pull request:
Most uses of Go, in data plane, are going to involve adding Go emitter configuration to an existing project. We'll cover `tsp init` generation in the azsdk-cli.

This file can also be used for any other tspconfig.yaml instructions as well if we want to try to cut down on github-instructions.md.

Using this prompt (via /tspconfig_add_missing_languages) look like this:

<img width="434" height="556" alt="image" src="https://github.com/user-attachments/assets/ed1315fc-3ac2-4c7f-bf00-3bb45e78eb1f" />

Fixes https://github.com/Azure/azure-sdk-tools/issues/11977

Commit:
Adding in some TypeSpec instructions to make it easy for someone to add in Go to an existing project.

richardpark-msft committed:
Merge branch 'main' into restapispecs-add-instructions-to-add-go

richardpark-msft committed:
Fixing a formatting bit for the 'flavor' line.

raych1 commented on .github/instructions/tspconfig.instructions.md:
```diff
@@ -0,0 +1,43 @@
+	service-dir: "{output-dir}/sdk/ai"
+    # take the final segment, prefix it with `az`, and make it all lowercase. 
+    # NOTE, this property builds on the one before it, so you don't have to add it again.
+	emitter-output-dir: "{service-dir}/azdocumentintelligence"
```
That's correct.

tadelesh commented on .github/instructions/tspconfig.instructions.md:
````diff
@@ -0,0 +1,43 @@
+Here's an example C# entry:
+
+```yaml
+"@azure-tools/typespec-csharp":
````
Go has different service dir and emitter output dir rule from other languages. Do we need to mention the .NET config which may cause mis-leading? I suggest it's better to show the template [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml#L42) and doc [here](https://azure.github.io/typespec-azure/docs/emitters/clients/typespec-go/reference/emitter/#emitter-options). Note that the template is not update to date before the [PR](https://github.com/Azure/azure-rest-api-specs/pull/37022) I'm working on merge.

richardpark-msft replied:
I can delete more of what's in there (I really only need the namespace attribute). What this enables is the automatic derivation of the Go config, using an existing .NET config in the same file. Just one less thing they have to look up and fix themselves, essentially.

tadelesh commented:
> @tadelesh, I didn't add in management plane instructions because I was unclear if I needed to - mgmt plane always seems supported on day 1 (vs data plane, which is always added after the fact)

I prefer also add mgmt. part here. New mgmt. service also needs to know how to add Go config. Mgmt. config template is [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L35).

Commit:
Remove unneeded fields

Commit:
Modifying this to contain less redundant information (ie, we want the template to the be the single source of truth)

Commit:
Merge branch 'restapispecs-add-instructions-to-add-go' of github.com:richardpark-msft/azure-rest-api-specs into restapispecs-add-instructions-to-add-go

Commit:
Converted my instructions to a prompt, instead. This avoids some issues:

- The instructions files (even when applied to specific file filters) are getting HUGE, and are only going to get worse if we keep tacking on stuff like this.
- I can specify a _specific_ model to use, which cuts down on testing A LOT. We use a lot of Claude, so I'm using Claude Sonnet 4.

richardpark-msft committed:
Merge branch 'main' into restapispecs-add-instructions-to-add-go

maririos commented on .github/prompts/tspconfig_add_missing_languages.prompt.md:
```diff
@@ -0,0 +1,14 @@
+---
+mode: agent
+model: Claude Sonnet 4
```
ohh I didn't know we could do this!!!!!!!

richardpark-msft replied:
Yeah, it is probably the best feature and it actually works! :)

maririos commented on .github/prompts/tspconfig_add_missing_languages.prompt.md:
```diff
@@ -0,0 +1,14 @@
+model: Claude Sonnet 4
+---
+
+Add a language that is missing (for instance, adding Go, Python, Javascript C# or Rust support), using these rules:
```
Are we sure we want to add Rust? @ronniegeraghty 

```suggestion
Add a language that is missing (for instance, adding Go, Python, Javascript, C#, or Rust support), using these rules:
```

maririos commented on .github/prompts/tspconfig_add_missing_languages.prompt.md:
```diff
@@ -0,0 +1,14 @@
+
+Add a language that is missing (for instance, adding Go, Python, Javascript C# or Rust support), using these rules:
+
+- New emitter configurations should be added as a child of the "options:" section, after any other emitters and their options.
```
should it start saying where the emitter configurations need to go? tspconfig.yaml? or it is not needed

richardpark-msft replied:
It does a good job of seeing the similarity between the two files, so I think we're okay.

tadelesh commented:
> > > > @tadelesh, I didn't add in management plane instructions because I was unclear if I needed to - mgmt plane always seems supported on day 1 (vs data plane, which is always added after the fact)
> > > 
> > > 
> > > I prefer also add mgmt. part here. New mgmt. service also needs to know how to add Go config. Mgmt. config template is [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L35).
> > 
> > 
> > Ha, as soon as I added in management plane support the "magic" stopped happening. I'll need to spend some more time to refine this.
> 
> Okay, I re-fixed it. I converted it to a prompt instead because it cuts down on the overall instructions file for the repo and lets you control some other aspects, like the model.

The prompt looks elegant.

maririos reviewed (commented):
Could you add description on how this would be use by the user? Is this something that we want people to be aware of that will require documentation somewhere?

Commit:
Remove Rust, add in oxford comma.

Commit:
Merge branch 'restapispecs-add-instructions-to-add-go' of github.com:richardpark-msft/azure-rest-api-specs into restapispecs-add-instructions-to-add-go

mikeharder commented:
First, should this file be added to `.github/prompts` or `.github/instructions`?  This repo moved from the former to the latter in this PR, and I don't think we should revert back, unless there is consensus.

https://github.com/Azure/azure-rest-api-specs/pull/34880/files

mikeharder reviewed (changes requested):
see comments

maririos commented:
> First, should this file be added to `.github/prompts` or `.github/instructions`? This repo moved from the former to the latter in this PR, and I don't think we should revert back, unless there is consensus.
> 
> https://github.com/Azure/azure-rest-api-specs/pull/34880/files
> 
> Once this is decided, how "protected" do we want this folder to be? Allow anyone with write access to the specs repo, or limit it to members of the "bypass" group (much smaller group with higher permissions)?

Thank you Mike! I was not aware of that :)
As for protection, I want to limit the folder to our team, so the "bypass" group. I have been adding myself to all the parths related to ai on this repo in codeowners so we can have control over what goes in.

@richardpark-msft We should then move it according to the GH Copilot documentation in the PR you linked.

mikeharder reviewed (changes requested):
change to v2 sample
